test/groovy/templates/PiperPipelineStageIntegrationTest.groovy

```diff
@@ -26,16 +29,34 @@ class PiperPipelineStageIntegrationTest extends BasePiperTest {
         helper.registerAllowedMethod('piperStageWrapper', [Map.class, Closure.class], {m, body ->
             return body()
         })
+
+        helper.registerAllowedMethod('npmExecuteScripts', [Map.class], {m ->
+            stepsCalled.add('npmExecuteScripts')
+            stepParameters.npmExecuteScripts = m
+        })
+
+        helper.registerAllowedMethod('testsPublishResults', [Map.class], {m ->
+            stepsCalled.add('testsPublishResults')
+            stepParameters.testsPublishResults = m
+        })
     }
 
     @Test
     void testStageDefault() {
-
         jsr.step.piperPipelineStageIntegration(
             script: nullScript,
             juStabUtils: utils,
         )
-        assertThat(jlr.log, containsString('No default stage implementation is provided for this stage.'))
+        assertThat(stepsCalled, not(hasItems('npmExecuteScripts', 'testsPublishResults')))
```

Comment:
Not sure about this assert. I guess `hasItems()` will return `true` if the called steps has *both* items and `false` also if one of them is present. So if you want to assert that neither step has been called, you may need two asserts, or rephrase the condition.

Reply:
Good catch. This is also wrong in all other tests. Will fix it there as well.